components/clm/src/biogeochem/CNPhenologyBeTRMod.F90

```diff
@@ -1542,45 +1576,78 @@ subroutine CropPhenology(num_pcropp, filter_pcropp                     , &
             else ! not winter cereal... slevis: added distinction between NH and SH
                ! slevis: The idea is that jday will equal idop sooner or later in the year
                !         while the gdd part is either true or false for the year.
-               if (t10(p) /= spval.and. a10tmin(p) /= spval   .and. &
-                    t10(p)     > planttemp(ivt(p))             .and. &
-                    a10tmin(p) > minplanttemp(ivt(p))          .and. &
-                    jday       >= minplantjday(ivt(p),h)       .and. &
-                    jday       <= maxplantjday(ivt(p),h)       .and. &
-                    t10(p) /= spval .and. a10tmin(p) /= spval  .and. &
-                    gdd820(p) /= spval                         .and. &
-                    gdd820(p) >= gddmin(ivt(p))) then
-
-                  ! impose limit on growing season length needed
-                  ! for crop maturity - for cold weather constraints
-                  croplive(p)  = .true.
-                  cropplant(p) = .true.
-                  idop(p)      = jday
-                  harvdate(p)  = NOT_Harvested
-
-                  ! go a specified amount of time before/after
-                  ! climatological date
-                  if (ivt(p)==nsoybean .or. ivt(p) == nsoybeanirrig) gddmaturity(p)=min(gdd1020(p),hybgdd(ivt(p)))
-                  if (ivt(p)==ncorn .or. ivt(p)==ncornirrig) then
-                     gddmaturity(p)=max(950._r8, min(gdd820(p)*0.85_r8, hybgdd(ivt(p))))
-                     gddmaturity(p)=max(950._r8, min(gddmaturity(p)+150._r8,1850._r8))
+               ! Assign seasonality - either temperature or precipitation
+               t_season = .false.
+               p_season = .false.
+               no_season = .false.
+               if (cvp(p) > tcvp) then        ! precipitation CV indicator
+                  if (cvt(p) >= tcvt) then    ! Both temperature and precip seasonality
+                     if (minval(xt_bar(p,:)) .lt. 283._r8)then ! cold season exists (i.e., min temp below 10 deg C)
```

Comment:
consider defining a variable for `283._r8`.

Reply:
OK